Assets/Scripts/Game/GameManager.cs

```diff
@@ -59,23 +61,43 @@ private IEnumerator HandleWallsDangerousness()
             {
                 yield return null;
 
-                while (!_waveManager.CurrentWave.switchOnWalls)
+                MoveRestingWallsToDangerous();
+
+                while (_waveManager.CurrentWave.dangerWallAmount == 0 || 
+                    _waveManager.CurrentWave.dangerWallAmount <= _dangerWallsCurrently ||
+                    _wallsToBeDangerous.Count == 0)
                 {
+                    MoveRestingWallsToDangerous();
```

Comment:
Looks like two situations are mixed here:
1. We don't need to make a dangerous wall (1 and 2 condition) => we just wait (don't need to call `MoveRestingWallsToDangerous`)
2. We have to make a wall dangerous but we just don't have free one => wait and continue the initial cycle. 

In both cases, the cycle is not needed, simple continue will do fine.

Reply:
Is it outdated comment or not?